### PR TITLE
Docs update: quarkus/spring instead of WildFly/Tomcat, CS instead of DRL, workbench/ES is dead, ...

### PR DESCRIPTION
**Pull request by ge0ffrey**

### CODE REVIEW

File: optaplanner-docs/src/main/asciidoc/OptimizationAlgorithms/OptimizationAlgorithms-chapter.adoc [hunk 1]
```diff
@@ -76,10 +76,17 @@ Do not assess quality on small data sets (unless production encounters only such
...
+with score calculation by a rule engine (such as Drools).
+This combination is very efficient, because:
+
+* A rule engine, such as Drools, is *great for calculating the score* of a solution of a planning problem.
+It makes it easy and scalable to add additional soft or hard constraints.
+It does <<incrementalScoreCalculation,incremental score calculation (deltas)>> without any extra code.
+However it tends to be not suitable to actually find new solutions.
+* An optimization algorithm is *great at finding new improving solutions* for a planning problem,
+without necessarily brute-forcing every possibility.
+However, it needs to know the score of a solution and offers no support in calculating that score efficiently.
```
Comment (triceo) [inline optaplanner-docs/src/main/asciidoc/OptimizationAlgorithms/OptimizationAlgorithms-chapter.adoc:89]: I'm not sure here. Do we need to mention Drools? In CS, it is an abstraction. It is a concept they do not need to know about. If things go well, users will never even notice it's there.

File: optaplanner-docs/src/main/asciidoc/PlannerIntroduction/DownloadAndRunTheExamples/DownloadAndRunTheExamples-section.adoc [hunk 2]
```diff
@@ -63,7 +63,7 @@ To run the examples in your favorite IDE:
 
 
 [[useWithMavenGradleEtc]]
-== Use OptaPlanner with Maven, Gradle, Ivy, Buildr, or ANT
+== Use OptaPlanner with Maven, Gradle, or ANT
```
Comment (triceo) [inline optaplanner-docs/src/main/asciidoc/PlannerIntroduction/DownloadAndRunTheExamples/DownloadAndRunTheExamples-section.adoc:66]: Are we keeping ANT then? I didn't notice any `build.xml` in our projects.

**Review by triceo — CHANGES_REQUESTED** [review]
Same comment about BAVET as on the other PR. Please remove the reference.

File: optaplanner-docs/src/main/asciidoc/OptimizationAlgorithms/OptimizationAlgorithms-chapter.adoc (hunk 1 above)
Comment (ge0ffrey) [reply optaplanner-docs/src/main/asciidoc/OptimizationAlgorithms/OptimizationAlgorithms-chapter.adoc:89]: Good point, I fully agree. This is the old text that I just cleaned up a bit and moved, I didn't refactor it along that far.

I'll make those changes.

File: optaplanner-docs/src/main/asciidoc/PlannerIntroduction/DownloadAndRunTheExamples/DownloadAndRunTheExamples-section.adoc (hunk 2 above)
Comment (ge0ffrey) [reply optaplanner-docs/src/main/asciidoc/PlannerIntroduction/DownloadAndRunTheExamples/DownloadAndRunTheExamples-section.adoc:66]: I was thinking of removing ANT too - nobody uses that any more - but we still need to talk about plain jar files that students drop in their IDE without maven or gradle.

**Comment by ge0ffrey** [comment]
Discussed, and agreed to merge this already

**Pull request merged**